Lib/test/test_fnmatch.py

```diff
@@ -250,6 +250,72 @@ def test_translate(self):
         self.assertTrue(re.match(fatre, 'cbabcaxc'))
         self.assertFalse(re.match(fatre, 'dabccbad'))
 
+    def test_translate_wildcards(self):
+        for pattern, expect in [
+            ('ab*', r'(?s:ab.*)\Z'),
+            ('ab*cd', r'(?s:ab.*cd)\Z'),
+            ('ab*cd*', r'(?s:ab(?>.*?cd).*)\Z'),
+            ('ab*cd*12', r'(?s:ab(?>.*?cd).*12)\Z'),
+            ('ab*cd*12*', r'(?s:ab(?>.*?cd)(?>.*?12).*)\Z'),
+            ('ab*cd*12*34', r'(?s:ab(?>.*?cd)(?>.*?12).*34)\Z'),
+            ('ab*cd*12*34*', r'(?s:ab(?>.*?cd)(?>.*?12)(?>.*?34).*)\Z'),
+        ]:
+            translated = translate(pattern)
+            self.assertEqual(translated, expect, pattern)
```

Comment:
Using a `for` loop like this makes the tracebacks slightly less useful when a test case fails. Either use a [subtest](https://docs.python.org/3/library/unittest.html#subtests) or spell each assertion out explicitly (e.g. `self.assertEqual(translate('ab*'), r'(?s:ab.*)\Z')`).

(Also below)

Reply:
I forgot to put the `subTest`. Thanks for reminding me of that!